server/src/models/ledger.rs

```diff
@@ -0,0 +1,137 @@
+use rust_decimal::Decimal;
+use serde::{Deserialize, Serialize};
+
+use crate::prelude::*;
+
+const LEDGER_BALANCE_SCALE: u32 = 9;
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub(crate) struct Ledger(pub(crate) Vec<LedgerAccount>);
+
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct LedgerAccount {
+    pub(crate) pk: String,
+    pub(crate) balance: String,
+    pub(crate) delegate: String,
+}
+
+impl Ledger {
+    pub(crate) async fn fetch(hash: impl Into<String>) -> Result<Ledger> {
+        let hash = hash.into();
+
+        let ledger = reqwest::get(f!(
+            "https://raw.githubusercontent.com/Granola-Team/mina-ledger/main/mainnet/{hash}.json"
+        ))
+        .await?
+        .bytes()
+        .await?;
+
+        Ok(Ledger(serde_json::from_slice(&ledger)?))
+    }
+
+    pub(crate) fn get_stake_weight(&self, public_key: impl Into<String>) -> Result<Decimal> {
+        let public_key = public_key.into();
+
+        let account = self
+            .0
+            .iter()
+            .find(|d| d.pk == public_key)
+            .ok_or_else(|| Error::Ledger(f!("account '{public_key}' not found in ledger.")))?;
```

Comment:
Can this error be converted to use `thiserror` then just pass in `Error::Ledger`?

Reply:
👍🏽